AppController/scripts/ejabberd_auth.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python2
+
+import hashlib
+import logging
+import os
+import re
+import socket
+import struct
+import sys
+import time
+
+from appscale.common import appscale_info
+from appscale.common.constants import LOG_FORMAT
+from appscale.common.ua_client import UAClient
+
+LOG_FILE = os.path.join('/', 'var', 'log', 'ejabberd', 'extauth.log')
+
+
+class EjabberdInputError(Exception):
+  def __init__(self, value):
+    self.value = value
+  def __str__(self):
+    return repr(self.value)
+
+
+def ejabberd_in():
+  logging.debug("trying to read 2 bytes from ejabberd:")
+  input_length = sys.stdin.read(2)
+
+  if len(input_length) is not 2:
+    logging.debug("ejabberd sent us wrong things!")
+    raise EjabberdInputError('Wrong input from ejabberd!')
+
+  logging.debug('got 2 bytes via stdin: %s'%input_length)
+  (size,) = struct.unpack('>h', input_length)
+
+  logging.debug('size of data: %i'%size)
+  income = sys.stdin.read(size).split(':')
```

Comment:
I'm not normally picky about variable names, but this one confused me for a second. `data` instead maybe?